xarray/backends/zarr.py

```diff
@@ -75,8 +89,10 @@ def __init__(self, zarr_array):
         self.shape = self._array.shape
 
         # preserve vlen string object dtype (GH 7328)
-        if self._array.filters is not None and any(
-            [filt.codec_id == "vlen-utf8" for filt in self._array.filters]
+        if (
```

Comment:
https://github.com/zarr-developers/zarr-python/pull/2036 is probably relevant here.

Reply:
Confirm whether we need any logic on the branch where we do have Zarr V3.

Reply:
I believe this is handled in the new `Numpy2StringDTypeCoder`.